syntax = "proto2";

/**
 * Open Source'd font metadata proto formats.
 *
 * This schema was determined empirically by observing
 * existing METADATA.pb files in the google/fonts repository.
 *
 */
package google.fonts;

option java_package = "com.google.fonts.proto";
option java_outer_classname = "FontsPublic";

message FamilyProto {
  required string name = 1;
  required string designer = 2;
  required string license = 3;
  required string category = 4;
  required string date_added = 5;
  repeated FontProto fonts = 6;
  repeated string aliases = 7;
  repeated string subsets = 8;
  optional string ttf_autohint_args = 9;
  repeated AxisSegmentProto axes = 10;
  map<string, float> registry_default_overrides = 11;
  optional SourceProto source = 12;
  bool is_noto = 13;
  repeated string languages = 14;
  repeated SampleGlyphsProto sample_glyphs = 15;
  repeated SampleTextProto sample_text = 16;
};

message SampleGlyphsProto {
  required string key = 1;
  required string value = 1;
}

message SampleTextProto {
  optional string masthead_full = 1;
  optional string masthead_partial = 2;
  optional string styles = 3;
  optional string tester = 4;
  optional string poster_sm = 5;
  optional string poster_md = 6;
  optional string poster_lg = 7;
}

message FontProto {
  required string name = 1;
  required string style = 2;
  required int32 weight = 3;
  required string filename = 4;
  required string post_script_name = 5;
  required string full_name = 6;
  optional string copyright = 7;
};

message AxisSegmentProto {
  optional string tag = 1;
  optional float min_value = 2;
  reserved 3;
  reserved "default_value";
  optional float max_value = 4;
};

message SourceProto {
  optional string repository_url = 1;
  optional string commit = 2;
}
